resources/services/network/virtual_networks_mock_test.go

```diff
@@ -88,6 +88,8 @@ func buildNetworkVirtualNetworksMock(t *testing.T, ctrl *gomock.Controller) serv
 	require.Nil(t, faker.FakeData(&(*vn.Subnets)[0].RouteTable.DisableBgpRoutePropagation))
 	require.Nil(t, faker.FakeData(&(*vn.Subnets)[0].RouteTable.ProvisioningState))
 	require.Nil(t, faker.FakeData(&(*vn.Subnets)[0].RouteTable.ResourceGUID))
+	require.Nil(t, faker.FakeData(&(*vn.Subnets)[0].IPConfigurations))
+	require.Nil(t, faker.FakeData(&(*vn.Subnets)[0].PrivateEndpoints))
```

Comment:
We need to skip these fields in the faker as they are recursive and are causing the tests to fail.
```suggestion
	require.Nil(t, faker.FakeDataSkipFields(&(*vn.Subnets)[0].IPConfigurations), []string{"PrivateEndpointProperties"})
	require.Nil(t, faker.FakeDataSkipFields(&(*vn.Subnets)[0].PrivateEndpoints), []string{"Subnet"})
```